pkgs/development/python-modules/osqp/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, pythonOlder
+, buildPythonPackage
+, fetchPypi
+, cmake
+, future
+, numpy
+  # check inputs
+, scipy
+, pytestCheckHook
+, mkl
+}:
+
+buildPythonPackage rec {
+  pname = "osqp";
+  version = "0.6.1";
+
+  disabled = pythonOlder "3.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "130frig5bznfacqp9jwbshmbqd2xw3ixdspsbkrwsvkdaab7kca7";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  dontUseCmakeConfigure = true;
+
+  propagatedBuildInputs = [
+    numpy
+    future
+  ];
+
+  checkInputs = [ scipy pytestCheckHook mkl ];
+  preCheck = ''
+    export LD_LIBRARY_PATH="${mkl}/lib/":$LD_LIBRARY_PATH;
```

Comment:
is mkl needed at runtime? or just part of the tests?

Reply:
As far as I can tell, MKL is an optional runtime dependency. https://github.com/oxfordcontrol/osqp/blob/d7eeb245d5b7f0195bfc35b6cb71f4570d7ca18b/docs/get_started/linear_system_solvers.rst

It comes with a system solver built in, but if MKL is available it uses it for faster/larger problems.